hfile/reader.go

```diff
@@ -254,3 +265,74 @@ func (r *Reader) GetIterator() *Iterator {
 		return NewIterator(r)
 	}
 }
+
+// TODO(benjy): Write tests for this, once the writer supports writing the FileInfo block.
+func (r *Reader) readFileInfo(data []byte) (err error) {
+	type outOfBounds struct {}
+
+	defer func() {
```

Comment:
We don't do this (recover OOBs) anywhere else -- I think I'd rather let it panic and get a useful stack trace.